sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -392,6 +392,156 @@ def get_access_control(self, upn=None,  # type: Optional[bool]
+        :return: A summary of the recursive operations, including the count of successes and failures,
+            as well as a continuation token in case the operation was terminated prematurely.
+        :rtype: :class:`~azure.storage.filedatalake.models.AccessControlChanges`
+        """
+        if not acl:
+            raise ValueError("The Access Control List must be set for this operation")
+
+        options = self._set_access_control_recursive_options('modify', acl=acl, **kwargs)
+        return self._set_access_control_internal(options, progress_callback)
+
+    def remove_access_control_recursive(self,
+                                        acl=None,  # type: Optional[str]
+                                        progress_callback=None,
+                                        **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Removes the Access Control on a path and sub-paths.
+
+        :param acl:
+            Removes POSIX access control rights on files and directories.
+            The value is a comma-separated list of access control entries. Each
+            access control entry (ACE) consists of a scope, a type, and a user or
+            group identifier in the format "[scope:][type]:[id]”.
+        :type acl: str
+        :param progress_callback:
+            Callback where the caller can track progress of the operation
+            as well as collect paths that failed to change Access Control.
+        :type progress_callback: func(~azure.storage.filedatalake._generated.models.SetAccessControlRecursiveResponse)
+        :keyword str continuation:
+            Optional continuation token that can be used to resume previously stopped operation.
+        :keyword int batch_size:
+            Optional. If data set size exceeds batch size then operation will be split into multiple
+            requests so that progress can be tracked. Batch size should be between 1 and 2000.
+            The default when unspecified is 2000.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: A summary of the recursive operations, including the count of successes and failures,
+            as well as a continuation token in case the operation was terminated prematurely.
+        :rtype: :class:`~azure.storage.filedatalake.models.AccessControlChanges`
+        """
+        if not acl:
+            raise ValueError("The Access Control List must be set for this operation")
+
+        options = self._set_access_control_recursive_options('remove', acl=acl, **kwargs)
+        return self._set_access_control_internal(options, progress_callback)
+
+    def _set_access_control_internal(self, options, progress_callback):
+        try:
+            total_directories_successful = 0
+            total_files_success = 0
+            total_failure_count = 0
+            continue_operation = True
+            while continue_operation:
```

Comment:
We should probably track the `timeout` value. i.e. if a user supplies a timeout value it should be for the entire operation, not individual calls - otherwise a user has no control over this loop.
Unless we expect it to be highly unlikely that this loop is more than one or two iterations?

Reply:
I added max_batch to enable the user to control the iteration more explicitly. Do you still think we need an overall timeout? I'm not sure that there's a scenario where the customer can accurately predict how long this operation can run for, as sometimes they have millions of files and it'll take hours to execute.